R/class-plotdata-histogram.R

```diff
@@ -46,50 +46,67 @@ newHistogramPD <- function(.dt = data.table::data.table(),
   group <- veupathUtils::toColNameOrNull(attr$overlayVariable)
   panel <- findPanelColName(attr$facetVariable1, attr$facetVariable2)
 
-  summary <- as.list(summary(.pd[[x]]))
-  names(summary) <- c('min', 'q1', 'median', 'mean', 'q3', 'max')
-  summary <- lapply(summary, as.character)
-  summary <- lapply(summary, jsonlite::unbox)
-  attr$summary <- summary
-  veupathUtils::logWithTime('Supporting summary statistics calculated for histogram.', verbose)
-
-  if (is.null(viewport)) {
-    viewport <- findViewport(.pd[[x]], xType)
-    veupathUtils::logWithTime('Determined default viewport.', verbose)
+  if (!length(.pd[[x]])) {
+    binSlider <- list('min'=jsonlite::unbox(NA), 'max'=jsonlite::unbox(NA), 'step'=jsonlite::unbox(NA))
+    binSpec <- list('type'=jsonlite::unbox(binReportValue), 'value'=jsonlite::unbox(NA))
+    viewport <- list('xMin'=0, 'xMax'=-Inf) 
+    summary <- list('min'=jsonlite::unbox(""),
```

Comment:
Why some empty strings and some NAs? Curious if there's a general guideline or something about the expected response would dictate one over the other.

Reply:
i think the rule weve agreed to is for scalar numbers to return null (js)/ NA (R), for scalar strings to return an empty one and for the main data essentially an empty frame of an object.

Reply:
i was talking w bob about it yesterday

Reply:
cool sounds good! Thank you!